app/models/post.rb

```diff
@@ -0,0 +1,18 @@
+class Post < ApplicationRecord
+  has_many :likes, foreign_key: 'posts_id'
+  has_many :comments, foreign_key: 'posts_id'
+  belongs_to :author, class_name: 'User', dependent: :destroy, foreign_key: 'author_id'
+
+  validates :title, presence: true
+  validates :title, length: { maximum: 250 }
+  validates :CommentsCounter, numericality: { greater_than_or_equal_to: 0 }
+  validates :LikesCounter, numericality: { greater_than_or_equal_to: 0 }
+
+  def update_posts_count
+    update(posts_count: posts.count)
+  end
```

Comment:
- [ ]  @shuhad786 it would have been nice if you ensured your post count is stored in the PostCounter, this will enable an easier display of post counts in your views file in future milestones.